src/TableFunctions/TableFunctionRedis.cpp

```diff
@@ -0,0 +1,77 @@
+
+#include <TableFunctions/TableFunctionRedis.h>
+#include <TableFunctions/TableFunctionFactory.h>
+#include <TableFunctions/parseColumnsListForTableFunction.h>
+#include "registerTableFunctions.h"
+#include <Interpreters/evaluateConstantExpression.h>
+#include <Interpreters/Context.h>
+
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Common/logger_useful.h>
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+
+StoragePtr TableFunctionRedis::executeImpl(const ASTPtr & /*ast_function*/,
+        ContextPtr context, const String & table_name, ColumnsDescription /*cached_columns*/) const
+{
+    auto columns = getActualTableStructure(context);
+    auto storage = std::make_shared<StorageRedis>( 
+    StorageID(getDatabaseName(), table_name), 
+    configuration->host,
+    configuration->port,
+    configuration->db_index,
+    configuration->password,
+    static_cast<RedisStorageType>(configuration->storage_type),
+    "primary_key",
+    "secondary_key",
+    "value",
+    context,
+    configuration->options,
+    columns,
+    ConstraintsDescription(),
+    String{});
+    storage->startup();
+    return storage;
+}
+
+ColumnsDescription TableFunctionRedis::getActualTableStructure(ContextPtr context) const
+{
+    String structure;
+    if (configuration->storage_type ==  StorageRedisConfiguration::RedisStorageType::HASH_MAP)
+        structure = "primary_key String, secondary_key String, value String";
```

Comment:
Maybe use terminology from Redis documentation: https://redis.io/commands/hget/ and use column names: `key`, `field`, `value`?